### PR TITLE
Add L2 penalty

### PR DESCRIPTION
**Pull request by brett-daley**
Important: if your network weights don't contain `'weights'` in their names, they won't be included in the calculation. Biases should not be included.

### CODE REVIEW

File: net.py [hunk 1]
```diff
@@ -84,15 +84,16 @@ def build_model(image_batch, batch_size, grayscale, reuse=False):
...
 
     if decay_lr:
         learning_rate, global_step = decaying_learning_rate(learning_rate)
     else:
         global_step = None
 
-    train_op = tf.train.GradientDescentOptimizer(learning_rate).minimize(loss, global_step=global_step)
+    train_op = tf.train.GradientDescentOptimizer(learning_rate).minimize(loss_with_l2_penalty, global_step=global_step)
 
     return loss, train_op
```
Comment (brett-daley) [inline net.py]: Note: this is **not** a bug here. I am intentionally returning the loss without the L2 penalty so we can easily compare it against the validation loss.
